### PR TITLE
Feature/auth

### PR DESCRIPTION
**Pull request by davidporosnicu**

### CODE REVIEW

File: src/container/Posts.js [hunk 1]
```diff
@@ -4,14 +4,16 @@ import { connect } from 'react-redux';
...
 
 class MainPage extends Component {
   componentDidMount() {
     this.props.fetchEntries();
   }
 
   render() {
-    const blogs = this.props.blogs;
+    const { blogs, auth } = this.props;
+    if (!auth.uid) return <Redirect to="/login" />;
```
Comment (horiaradu) [inline src/container/Posts.js]: Good try. I have a suggestion.

* create a higher order component which, checks if you are logged in
* if you are, then it renders the component you gave it
* if you are not, it redirects you to login
* put that HOC, with Posts as a parameter, on the route

In this way, if you have multiple protected routes, you don't copy paste this code everywhere.

File: src/redux/actions/authAuctions.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+export const signIn = userCredentials => (dispatch, getState, { getFirebase }) => {
```
Comment (horiaradu) [inline src/redux/actions/authAuctions.js:1]: What is `getFirebase` here?

File: src/redux/reducers/index.js [hunk 3]
```diff
@@ -1,6 +1,10 @@
 import { combineReducers } from 'redux';
 import blogReducer from './blogReducers';
+import authReducer from './authReducers';
+import { firebaseReducer } from 'react-redux-firebase';
 
 export default combineReducers({
-  blog: blogReducer
+  firebase: firebaseReducer,
```
Comment (horiaradu) [inline src/redux/reducers/index.js:7]: What does this firebase reducer do?

File: src/store.js [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
-import { reactReduxFirebase } from 'react-redux-firebase';
+import { reduxFirestore, getFirestore } from 'redux-firestore';
+import { reactReduxFirebase, getFirebase } from 'react-redux-firebase';
 import fbConfig from './config/fbConfig';
 
 const store = createStore(
   rootReducer,
   compose(
-    applyMiddleware(thunk),
+    applyMiddleware(thunk.withExtraArgument({ getFirebase, getFirestore })),
```
Comment (horiaradu) [inline src/store.js:11]: What is this middleware doing?

**Review by horiaradu — CHANGES_REQUESTED** [review]

File: src/store.js (hunk 4 above)
Comment (davidporosnicu) [reply src/store.js:11]: it allows us to pass an extra argument to our actions creators. which in that case is an object with two properties: getFirebase and getFirestore.

Comment (davidporosnicu) [reply src/store.js:11]: and we do that to get access to the firebase api. essentially its just an extra package for redux to work 'better' with firebase

File: src/redux/actions/authAuctions.js (hunk 2 above)
Comment (davidporosnicu) [reply src/redux/actions/authAuctions.js:1]: getFirebase is a method that comes from 'react-redux-firebase' package and allows me access informations from firebase... to get data or to send data. in this case i was sending data to check the credentials for that specific user to see if there is an user with that email and password.

File: src/redux/reducers/index.js (hunk 3 above)
Comment (davidporosnicu) [reply src/redux/reducers/index.js:7]: firebaseReducer also comes from 'react-redux-firebase' package and it allows me to access data about a specific user. for example: the id of the user.

File: src/container/Posts.js (hunk 1 above)
Comment (davidporosnicu) [reply src/container/Posts.js]: done

**Review by horiaradu — APPROVED** [review]

**Pull request merged**